detectorCommon/src/xtalk_neighbors_moduleThetaMergedSegmentation.cpp

```diff
@@ -0,0 +1,245 @@
+	(theta_id_neighbour > aFieldExtremes_layer[layer_indice_offset][idThetaField].second)
+	) continue;
+    // set theta_id of cell ID
+    aDecoder[aSeg.fieldNameTheta()].set(cID, theta_id + i*aSeg.mergedThetaCells(layer_id));
+    xtalk_neighbours.emplace_back(std::make_pair(cID, dummy_xtalk_theta));
+  }
+ 
+  // cross talk type 4
+  // reset cellID
+  aDecoder.set(cID, aSeg.fieldNameLayer(), layer_id);
+  aDecoder.set(cID, aSeg.fieldNameTheta(), theta_id);
+  for (int loopLayer = aFieldExtremes_layer[0][idLayerField].first; loopLayer<=aFieldExtremes_layer[0][idLayerField].second; loopLayer++) {
+    // cells in adjacent layers have been studied, skip
+    if ((loopLayer - layer_id) <= 1 && (loopLayer - layer_id) >= -1) continue;
+    aDecoder.set(cID, aSeg.fieldNameLayer(), loopLayer);
+    int loopLayer_indice_offset = loopLayer - aFieldExtremes_layer[0][idLayerField].first;
+    // find crosstalk neighbours in the same theta tower
+    for (int i=-1; i <= aSeg.mergedThetaCells(layer_id); i++) {
+      int theta_id_neighbour = (theta_id + i) - ((theta_id + i) % (aSeg.mergedThetaCells(loopLayer) ? aSeg.mergedThetaCells(loopLayer):1) );
+      // check if the neighbour theta index is valid
+      if (theta_id_neighbour >= theta_id && theta_id_neighbour < (theta_id + aSeg.mergedThetaCells(layer_id)) && theta_id_neighbour>=aFieldExtremes_layer[loopLayer_indice_offset][idThetaField].first && theta_id_neighbour<=aFieldExtremes_layer[loopLayer_indice_offset][idThetaField].second) { // crosstalk neighbour type 4
+        dummy_xtalk = dummy_xtalk_tower;
+        aDecoder.set(cID, aSeg.fieldNameTheta(), theta_id_neighbour);
+        xtalk_neighbours.emplace_back(std::make_pair(cID, dummy_xtalk));
+      }
+      else if (theta_id_neighbour < theta_id && theta_id_neighbour > (theta_id - aSeg.mergedThetaCells(loopLayer)) && theta_id_neighbour>=aFieldExtremes_layer[loopLayer_indice_offset][idThetaField].first && theta_id_neighbour<=aFieldExtremes_layer[loopLayer_indice_offset][idThetaField].second) { // crosstalk neighbour type 4
+        dummy_xtalk = dummy_xtalk_tower;
+        aDecoder.set(cID, aSeg.fieldNameTheta(), theta_id_neighbour);
+        xtalk_neighbours.emplace_back(std::make_pair(cID, dummy_xtalk));
+      }
+      else continue;
+    }
+  }
+
+  // remove duplicates
+  std::vector<uint64_t> sort_neighbours;
+  std::vector<std::pair<uint64_t, double>> new_neighbours;
+  for (auto &i : xtalk_neighbours)
+  {
+    bool IsNewNeighbour=true;
+    for (unsigned int i_loop=0; i_loop<sort_neighbours.size(); i_loop++)
+    {
+      if (sort_neighbours[i_loop]==i.first)
+      {
+        IsNewNeighbour=false;
+      }
+    }
+    if (IsNewNeighbour)
+    {
+      new_neighbours.emplace_back(i);
+      sort_neighbours.emplace_back(i.first);
+    }
+  }
+  xtalk_neighbours=new_neighbours;
+
+  return xtalk_neighbours;
+}
+
+// return cell position, a function used for debug purpose
```

Comment:
This does not return a position, it returns the indexes of the corresponding layer/module/theta fields.

Reply:
Changed from "cell position" to "indices of layer/module/theta". Similar change has also been implemented in a comment of xtalk_neighbors_moduleThetaMergedSegmentation.h
https://github.com/zwu0922/k4geo/blob/main/detectorCommon/include/detectorCommon/xtalk_neighbors_moduleThetaMergedSegmentation.h#L46

Reply:
Thanks! Could you also change the name of the function `xtalk_get_cell_position` --> `xtalk_get_cell_indices` if not too painful?

Reply:
Done. The change of the function name has been also implemented in k4RecCalorimeter accordingly.